hipchat.go

```diff
@@ -86,15 +86,22 @@ type ErrorResponse struct {
 type Client struct {
 	AuthToken string
 	BaseURL   string
+	ProxyURL  string
```

Comment:
+ Add the `Timeout` and `Transport` properties to the Client struct.
+ In each instance where `http.Get` or `http.PostForm` is used, do something like this:
```Go
 	req, err := http.NewRequest(...)
 	if err != nil {
 		// Handle error
 	}
  	// Create the HTTP client
  	client := &http.Client{
		Transport: c.Transport,
  		Timeout:   c.Timeout,
  	}
  
 	// Make the request
 	res, err := client.Do(req)
 	if err != nil {
 		// Handle error
 	}
```